lib/fastlane/plugin/revenuecat_internal/actions/git_clone_and_push.rb

```diff
@@ -0,0 +1,40 @@
+require 'fastlane/action'
+require 'fastlane_core/configuration/config_item'
+require_relative '../helper/revenuecat_internal_helper'
+require_relative '../helper/versioning_helper'
+
+module Fastlane
+  module Actions
+    class GitCloneAndPushAction < Action
+      def self.run(params)
+        source_repo = params[:source_repo]
+        destination_repo = params[:destination_repo]
+
+        Helper::RevenuecatInternalHelper.git_clone_source_to_dest(source_repo, destination_repo)
+      end
+
+      def self.description
+        "Makes a lightweight clone containing only on the main branch's history."
+      end
+
+      def self.authors
+        ["James Borthwick"]
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :source_repo,
+                                       description: "The source repository URL",
+                                       is_string: true),
```

Comment:
I don't remember if `type: String` is also required (I see it in the other actions). But if it works, I think it should be ok.

Reply:
Apparently `is_string` is an outdated way of doing this. I've replaced it with `type: String`, and added `optional: false` as well.